client/verta/verta/registry/entities/_modelversion.py

```diff
@@ -1514,3 +1516,98 @@ def set_hide_output_label(self, hide):
     def get_hide_output_label(self):
         self._refresh_cache()
         return self._msg.hide_output_label
+
+    def log_dataset(self, key, path, path_only, artifact_type, linked_artifact_id):
+        """
+        Logs a dataset to this Model Version.
+
+        Parameters
+        ----------
+        key : str
+            Artifact key.
+        path : str
+            Artifact path.
+        path_only : bool
+            Whether path only.
+        artifact_type : CommonService_pb2.ArtifactTypeEnum variant
+            Category to which the artifact belongs (e.g. MODEL, DATA).
+        linked_artifact_id : str
+            Linked artifact id.
+
+        """
+        if not key:
+            raise ValueError("key is not specified")
+        if not path:
+            raise ValueError("artifact_path is not specified")
+        if not path_only:
+            raise ValueError("path_only is not specified")
+        if not artifact_type:
+            raise ValueError("artifact_type is not specified")
+        if not linked_artifact_id:
+            raise ValueError("linked_artifact_id is not specified")
```

Comment:
I do not think these checks are necessary, because Python will require all of these arguments when a user calls `log_dataset()`.